### PR TITLE
Rework of publisher specification, spec reorganization, media tests

### PR DESCRIPTION
**Pull request by ryanml**
Continue off #18

### CODE REVIEW

File: js/publisher.js [hunk 1]
```diff
@@ -24,24 +26,19 @@ class JS extends Publisher {
     this.loadStubs(['publisher'])
   }
 
-  beforeEach (mockery) {
-    this.ledger.setSynopsis(null)
+  runBeforeEach (mockery) {
+    this.beforeEach(mockery)
+    this.changeSetting('PAYMENTS_ALLOW_NON_VERIFIED', false)
```
Comment (jasonrsadler) [inline js/publisher.js]: Just curious, why are we not allowing non-verified here?

Comment (ryanml) [reply js/publisher.js]: This was an oversight, removing, thanks for catching.

Comment (ryanml) [reply js/publisher.js]: fixed

File: js/spec.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        sinon.stub(this.ledger, stub.name).callsFake(stub.func.bind(...args))
+      })
+    })
+  }
+
+  setState (newState) {
+    let oldState = this.state
+    this.state = newState
+
+    if (!newState.hasIn(['ledger', 'info'])) {
```
Comment (jasonrsadler) [inline js/spec.js]: Should we do isImmutable check here inside the if?

Comment (ryanml) [reply js/spec.js]: I have not run in to such a scenario but probably a good addition

**Pull request merged**